tests/integration_tests/functional/test_balloon.py

```diff
@@ -0,0 +1,713 @@
+    ssh_connection = net_tools.SSHConnection(microvm.ssh_config)
+
+    # Get the firecracker from snapshot pid, and open an ssh connection.
+    firecracker_pid = microvm.jailer_clone_pid
+
+    # Check memory usage.
+    third_reading = get_rss_mem_by_pid(firecracker_pid)
+
+    # Dirty 60MB of pages.
+    make_guest_dirty_memory(ssh_connection, amount=(60 * MB_TO_PAGES))
+    time.sleep(1)
+
+    # Check memory usage.
+    fourth_reading = get_rss_mem_by_pid(firecracker_pid)
+
+    assert fourth_reading > third_reading
+
+    # Inflate the balloon with another 20MB of pages.
+    response = microvm.balloon.patch(amount_mb=40)
+    assert microvm.api_session.is_status_no_content(response.status_code)
+    time.sleep(5)
+
+    fifth_reading = get_rss_mem_by_pid(firecracker_pid)
+
+    # There should be a reduction in RSS, but it's inconsistent.
+    # We only test that the reduction happens.
+    assert fourth_reading > fifth_reading
+
+    microvm.kill()
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_snapshot_compatibility(
+    network_config,
+    bin_cloner_path
+):
+    """Test that the balloon serializes correctly."""
+    logger = logging.getLogger("snapshot_sequence")
+
+    # Create the test matrix.
+    test_matrix = build_test_matrix(network_config, bin_cloner_path, logger)
+
+    test_matrix.run_test(_test_snapshot_compatibility)
+
+
+def _test_snapshot_compatibility(context):
+    logger = context.custom['logger']
+    vm_builder = context.custom['builder']
+    snapshot_type = context.custom['snapshot_type']
+    enable_diff_snapshots = snapshot_type == SnapshotType.DIFF
+
+    logger.info("Testing {} with microvm: \"{}\", kernel {}, disk {} "
+                .format(snapshot_type,
+                        context.microvm.name(),
+                        context.kernel.name(),
+                        context.disk.name()))
+
+    # Create a rw copy artifact.
+    root_disk = context.disk.copy()
+    # Get ssh key from read-only artifact.
+    ssh_key = context.disk.ssh_key()
+    # Create a fresh microvm from aftifacts.
+    microvm = vm_builder.build(
+        kernel=context.kernel,
+        disks=[root_disk],
+        ssh_key=ssh_key,
+        config=context.microvm,
+        enable_diff_snapshots=enable_diff_snapshots
+    )
+
+    # Add a memory balloon with stats enabled.
+    response = microvm.balloon.put(
+        amount_mb=0,
+        deflate_on_oom=True,
+        must_tell_host=False,
+        stats_polling_interval_s=1
+    )
+    assert microvm.api_session.is_status_no_content(response.status_code)
+
+    microvm.start()
+
+    logger.info("Create {} #0.".format(snapshot_type))
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(microvm)
+
+    # Create a snapshot with a balloon on version 0.23.0.
+    # This should fail as the balloon was introduced in 0.24.0.
+    try:
+        snapshot_builder.create(
```

Comment:
Does this test fail if this create does not fail as it should?

Reply:
You're right, it didn't. It does now.